core/src/main/scala/kafka/coordinator/group/GroupMetadataManager.scala

```diff
@@ -82,19 +82,38 @@ class GroupMetadataManager(brokerId: Int,
 
   this.logIdent = "[Group Metadata Manager on Broker " + brokerId + "]: "
 
-  newGauge("NumOffsets",
+  private def recreateGauge[T](name: String, gauge: Gauge[T]): Gauge[T] = {
```

Comment:
Do we really need this? Can we not do the clean up in the test? If we think this is important, we should probably do it for all the gauges in `KafkaMetricsGroup`, but I am not sure if we need it.